Add doc about Test Runner iframe. (#3073)
cypress-io/cypress-documentation: source/guides/core-concepts/test-runner.md

{% imgTag /img/guides/errors.png "Errors" %}

> *Note: Internally, AUT uses iframe. It can sometimes cause unexpected {% url "behaviors like this." /api/commands/window.html#Cypress-uses-2-different-windows %}*

Reviewer: ```suggestion
*Note: Internally, the AUT renders within an iframe. This can sometimes cause unexpected behaviors {% url "explained here." /api/commands/window.html#Cypress-uses-2-different-windows %}*
```
